drivers/adc/adc_ads1x4s0x.c

```diff
@@ -18,400 +18,406 @@
 
 #define ADC_CONTEXT_USES_KERNEL_TIMER 1
 #define ADC_CONTEXT_WAIT_FOR_COMPLETION_TIMEOUT                                                    \
-	K_MSEC(CONFIG_ADC_ADS114S0X_WAIT_FOR_COMPLETION_TIMEOUT_MS)
+	K_MSEC(CONFIG_ADC_ADS1X4S0X_WAIT_FOR_COMPLETION_TIMEOUT_MS)
 #include "adc_context.h"
 
-LOG_MODULE_REGISTER(ads114s0x, CONFIG_ADC_LOG_LEVEL);
+LOG_MODULE_REGISTER(ads1x4s0x, CONFIG_ADC_LOG_LEVEL);
 
-#define ADS114S0X_CLK_FREQ_IN_KHZ                           4096
-#define ADS114S0X_RESET_LOW_TIME_IN_CLOCK_CYCLES            4
-#define ADS114S0X_START_SYNC_PULSE_DURATION_IN_CLOCK_CYCLES 4
-#define ADS114S0X_SETUP_TIME_IN_CLOCK_CYCLES                32
-#define ADS114S0X_INPUT_SELECTION_AINCOM                    12
-#define ADS114S0X_RESOLUTION                                16
-#define ADS114S0X_REF_INTERNAL                              2500
-#define ADS114S0X_GPIO_MAX                                  3
-#define ADS114S0X_POWER_ON_RESET_TIME_IN_US                 2200
-#define ADS114S0X_VBIAS_PIN_MAX                             7
-#define ADS114S0X_VBIAS_PIN_MIN                             0
+#define ADS1X4S0X_CLK_FREQ_IN_KHZ                           4096
```

Comment:
I'm definitely sure that it would be easier to review this if the renaming (especially all the defines, ... etc.) would be in a separate commit, because the actual changes are easily missed in this way.

Reply:
My apology... It was too late before I actually realized this.

Reply:
It is rather difficult for me to figure out the actual changes within all the renaming. But as I'm neither a collaborator nor a maintainer I don't want to enforce something which is not required.

Reply:
@anangl @henrikbrixandersen Do you think the renaming should be separated? And should each commit by itself successfully build, therefore removing the necessity of squashing?

Reply:
So here is the problem: If I rename a file but also change its content, there is a good chance that git cannot pick it up as a rename but instead treat it as a deletion and addition. This is especially troublesome since the name changes inside the file are more than 50% of the file itself.

What I can do is consolidate all name changes inside a single commit, so the the code will function exactly the same. But the price is that commit will be rather unreadable (as you will see the deletion of a file and the addition of a new file but they are essentially the same). If you would trust me that the only thing I did in that commit is renaming, I will be happy to do so 😄

Reply:
`git mv` ?

Reply:
Nope... I tried that.

Proposal: I do a commit that renames all variables first, then do a second commit that renames the files (and references to those files), just for clarity. These two rename commits will be squashed after this PR is approved, so it will be a large rename commit and should be able to build.

Reply:
Fine for me, but I also do not think the squash is necessary.

Reply:
Done. Now all these rename commits should be able to compile.

To rebase to not rebase, it's a problem and I shall leave to others to decide ;)

Reply:
done